### PR TITLE
feat: Re-enable typescript for production builds

### PR DESCRIPTION
**Pull request by marlonkeating**
This is a fixed version of https://github.com/openedx/frontend-build/pull/294 (reverted in https://github.com/openedx/frontend-build/pull/315), with package.json that includes typescript libraries in dependencies instead of devDependencies, which ensures that the requires typescript libraries will get pulled into any consumers of openedx/frontend-build.

### Example App
- [x] npm run build succeeds 
- [x] npm run test succeeds
### Local MFE
using typescript pulling frontend-build but no typescript libraries
- [x] npm run build succeeds 
- [x] npm run test succeeds

### CODE REVIEW

**Comment by abdullahwaheed** [comment]
Thank you @marlonkeating for updating this and fixing the issues.  Since now we have separate repo for [typescript-config](https://github.com/openedx/typescript-config), i think we should move all the typescript related configurations in that repo and just update the respective changes here.
cc: @adamstankiewicz

File: example/src/App.jsx [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import config from 'env.config';
+import Image from './Image.tsx';
```
Comment (adamstankiewicz) [inline example/src/App.jsx:2]: Shouldn't need to file extensions for ESLint.

File: example/src/Image.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+type ImageProps = {
+  src: string;
+  alt?: string;
+  style?: CSSProperties;
+};
+
+export default function Image(props:ImageProps) {
+  return <img {...props} />;
+}
```
Comment (adamstankiewicz) [inline example/src/Image.tsx:11]: nit: newline

File: tsconfig.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+      "sourceMap": true,
+      "strict": true,
+      "strictFunctionTypes": false,
+      "target": "ES6",
+      "rootDir": "example",
+      "outDir": "dist"
+    },
+    "include": ["example/**/*"],
+    "exclude": ["example/dist/*", "example/node_modules/*'"]
+  }
```
Comment (adamstankiewicz) [inline tsconfig.json:27]: nit: newline

**Review by adamstankiewicz — COMMENTED** [review]
CI is failing on `npm run lint` due to omitted `.tsx` file extensions. I believe we need to modify our ESLint config to add TypeScript support so it doesn't complain about things like this.

This can be achieved with `airbnb-typescript`, but would recommend adding it to our shared `@edx/eslint-config` library, given that's where our ESLint config extends `airbnb` and `airbnb/hooks` already.

See this PR: https://github.com/openedx/eslint-config/pull/124

File: package.json [hunk 4]
```diff
@@ -73,6 +77,8 @@
     "sharp": "^0.31.0",
     "source-map-loader": "^4.0.1",
     "style-loader": "3.3.2",
+    "ts-jest": "^26.5.0",
+    "typescript": "^4.9.4",
```
Comment (adamstankiewicz) [inline package.json:81]: [clarification] Should `typescript` be removed from the `devDependencies` given its now installed in `dependencies`?

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  "config/.eslintrc.js",
+  "example/.eslintrc.js",
+  "example/**/*",
+  "lib",
+  "bin",
+  "config"],
+  "exclude": ["node_modules",
+  "dist",
+  "example/dist/*",
+  "example/node_modules/*'"]
```
Comment (adamstankiewicz) [inline tsconfig.json:17]: nit: indenting (or lack thereof) makes this difficult to read at first glance:

```js
{
  "include": [
    ".eslintrc.js",
    "config/.eslintrc.js",
    "example/.eslintrc.js",
    // ...
  ]
}
```

File: example/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+  "include": [
+    ".eslintrc.js",
+    "env.config.js",
+    "src"
+  ],
+  "exclude": [
+    "node_modules",
+    "dist",
+  ]
+}
```
Comment (adamstankiewicz) [inline example/tsconfig.json:16]: nit: newline needed

File: example/src/Image.tsx [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+};
+
+const Image = ({ alt, ...rest }:ImageProps) => <img alt={alt} {...rest} />;
+
+const defaultProps = {
+  alt: undefined,
+  style: undefined,
+};
+Image.defaultProps = defaultProps;
+export default Image;
```
Comment (adamstankiewicz) [inline example/src/Image.tsx:16]: nit: newline needed

**Review by adamstankiewicz — APPROVED** [review]
LGTM to merge into `alpha`, though with a clarifying question first about the `typescript` dependency and a few nits.

**Pull request merged**